build/begin_addon_release.yml

```diff
@@ -34,18 +34,26 @@ stages:
         displayName: Setup Azure CLI
 
       - bash: |
-            # Greps the PR title for buildNum. -o returns just the matching results in the regex not the entire line, -P enables Perl regex matching, \K ignores anything before it in the final output, so the result is only what is inside the brackets 
-            nativeBuildNum=$(echo "$(Build.SourceVersionMessage)" | grep -o '\[buildNum:[^]]*\]' | grep -oP '\[buildNum:\K[^]]+')
-            itwinjsPR=$(echo "$(Build.SourceVersionMessage)" | grep -o '\[itwinjsPR:[^]]*\]' | grep -o '[0-9]\+')
+            commitMsg="$(Build.SourceVersionMessage)"
+
+            # Use grep to parse commit message for PR number attached to end of the commit
+            # grep -o is for return only the matched pattern, -E is for expanded regex syntax for [0-9]+
+            nativePrNum=$(echo "$commitMsg" | grep -oE '\(#([0-9]+)\)$' | grep -o '[0-9]\+')
+            echo "Native PR Num: $nativePrNum"
+
+            # Store PR description in txt file because vars behave strangely with multi line comments
+            gh pr view $nativePrNum --json body  --repo iTwin/imodel-native-internal -q '.body' >> body.txt
+
+            # parse body.txt seraching for itwinjs-core pr URl and extract pull number from line. grep -o ensure the grep only returns the matching pattern instead of the entire line
+            itwinjsPR=$(grep -o '^itwinjs-core: https://github.com/iTwin/itwinjs-core/pull/[0-9]\+' body.txt | grep -o '[0-9]\+')
+            echo "itwinjs-core pr Num $itwinjsPR"
 
             if [[ -z "$itwinjsPR" ]]; then
-              iTwinjsBranchName='master'
-            else
-              iTwinjsBranchName=$(gh pr view $itwinjsPR --json headRefName  --repo iTwin/itwinjs-core -q '.headRefName')
+              itwinjsPR='0'
+              echo "No itwinjs-core PR found. Defaulting PR num to: $itwinjsPR"

```

Comment:
Whatever additional comment you add to the other 0 default should probably go here too.